apps/src/templates/textResponses/TextResponses.jsx

```diff
@@ -23,150 +24,120 @@ const CSV_HEADERS = [
   {label: i18n.question(), key: 'question'},
   {label: i18n.response(), key: 'response'}
 ];
-const DEFAULT_FILTER_KEY = i18n.all();
 const PADDING = 8;
 
-class TextResponses extends Component {
-  static propTypes = {
-    // Provided by redux.
-    sectionId: PropTypes.number.isRequired,
-    responses: PropTypes.object.isRequired,
-    isLoadingResponses: PropTypes.bool.isRequired,
-    validScripts: PropTypes.arrayOf(validScriptPropType).isRequired,
-    scriptId: PropTypes.number,
-    scriptName: PropTypes.string,
-    setScriptId: PropTypes.func.isRequired,
-    asyncLoadTextResponses: PropTypes.func.isRequired
-  };
-
-  state = {
-    filterByLessonName: null
-  };
-
-  componentDidMount() {
-    this.props.asyncLoadTextResponses(
-      this.props.sectionId,
-      this.props.scriptId
-    );
-  }
-
-  getResponsesByScript = () => {
-    const {responses, scriptId} = this.props;
-    return responses[scriptId] || [];
-  };
-
-  onChangeScript = scriptId => {
-    const {setScriptId, asyncLoadTextResponses, sectionId} = this.props;
-    asyncLoadTextResponses(sectionId, scriptId, () => {
-      setScriptId(scriptId);
-      this.setState({filterByLessonName: null});
-    });
-  };
-
-  renderFilterByLessonDropdown = () => {
-    const lessons = this.getLessons();
-
-    // only render filter dropdown if there are 2+ lessons
-    if (lessons.length <= 1) {
-      return null;
+function TextResponses({
+  sectionId,
+  validScripts,
+  scriptId,
+  scriptName,
+  setScriptId
+}) {
+  const [textResponsesByScript, setTextResponsesByScript] = useState({});
+  const [isLoadingResponses, setIsLoadingResponses] = useState(false);
+  const [filterByLessonName, setFilterByLessonName] = useState(null);
+
+  useEffect(() => {
+    asyncLoadTextResponses(sectionId, scriptId);
+  }, [scriptId]);
+
+  const asyncLoadTextResponses = (sectionId, scriptId) => {
+    // Don't load data if it's already stored in state.
+    if (textResponsesByScript[scriptId]) {
```

Comment:
Do we need to also track `sectionId` in `textResponsesByScript` somehow?

Reply:
The way teacher dashboard is right now, section ID doesn't change without a full page reload

Reply:
I see.  I'm a little nervous that we'll trip over this limitation later, could we at least add a comment stating this assumption somewhere?

Reply:
Actually, I'll add some logic that will reset the textResponsesByScript if the sectionId changes, which right now will never get called, but would future-proof the component.